admin-dev/themes/new-theme/js/pages/product/edit/combinations-manager.js

```diff
@@ -27,37 +27,119 @@ import ProductMap from '@pages/product/product-map';
 import CombinationsGridRenderer from '@pages/product/edit/combinations-grid-renderer';
 import CombinationsService from '@pages/product/services/combinations-service';
 import DynamicPaginator from '@components/pagination/dynamic-paginator';
+import SubmittableInput from '@components/form/submittable-input';
 
 const {$} = window;
 
 export default class CombinationsManager {
   constructor() {
     this.$productForm = $(ProductMap.productForm);
+    this.$combinationsContainer = $(ProductMap.combinations.combinationsContainer);
+    this.combinationIdInputsSelector = ProductMap.combinations.combinationIdInputsSelector;
     this.initialized = false;
+    this.combinationsService = new CombinationsService(this.getProductId());
+
     this.init();
 
     return {};
   }
 
+  /**
+   * @private
+   */
   init() {
-    const productId = this.getProductId();
     this.paginator = new DynamicPaginator(
       ProductMap.combinations.paginationContainer,
-      new CombinationsService(productId),
+      this.combinationsService,
       new CombinationsGridRenderer(),
     );
+    this.initSubmittableInputs();
+    this.$combinationsContainer.on('change', ProductMap.combinations.isDefaultInputsSelector, (e) => {
+      if (!e.currentTarget.checked) {
+        return;
+      }
+      this.updateDefaultCombination(e.currentTarget);
+    });
 
     // Paginate to first page when tab is shown
     this.$productForm.find(ProductMap.combinations.navigationTab).on('shown.bs.tab', () => this.firstInit());
   }
 
+  /**
+   * @private
+   */
+  initSubmittableInputs() {
+    const combinationToken = this.getCombinationToken();
+    const {quantityKey} = ProductMap.combinations.combinationItemForm;
+    const {impactOnPriceKey} = ProductMap.combinations.combinationItemForm;
+    const {referenceKey} = ProductMap.combinations.combinationItemForm;
+    const {tokenKey} = ProductMap.combinations.combinationItemForm;
+
+    new SubmittableInput(ProductMap.combinations.quantityInputWrapper, async (input) => {
```

Comment:
The handle of the Promise in the component seems good, however the provided callback when the components element doesn't seem to be a Promise. Because `updateListedCombination` does return a Promise but it doesn't seem the Promise is returned here

So I wonder how it can work 🤔  I might be wrong but un would expect a `return this.combinationsService.updateListedCombination`

Reply:
I tested the PR and everything works fine indeed I don't pretend I fully understand the error case with Promises despite @NeOMakinG explanations 😅
But anyway it's fine!